debug-ui/scripts/stats-server.ts

```diff
@@ -1172,11 +1193,24 @@ export class ManifestStatsServer {
       await client.query('COMMIT');
       console.log('State saved successfully to database');
     } catch (error) {
-      await client.query('ROLLBACK');
       console.error('Error saving state to database:', error);
-      throw error;
+      if (client) {
+        try {
+          await client.query('ROLLBACK');
+        } catch (rollbackError) {
+          console.error('Error during rollback:', rollbackError);
+        }
+      }
+      // Don't throw here - log error but allow server to continue
```

Comment:
keep the throw. it is caught a level above. dont mix where errors are caught unless you are doing something different